### DISCUSSION TITLE
REPL over UART only

### DISCUSSION THREAD

**Discussion by hugolundin** [micropython/micropython#13753] [RP2040 / Pico]
I want to use the USB serial device for sending raw bytes with pyserial only, and not have the REPL there. I tried setting `MICROPY_HW_ENABLE_UART_REPL` to `(1)` in `mpconfigport.h`, but the REPL is still available over USB serial.

Can this by done, either in `boot.py` (dupterm?) or via `mpconfigport.h`? Thank you!

**Comment by msawicki-poland** [comment]
Look at my attempt and solution, I build Micropython: https://forums.raspberrypi.com/viewtopic.php?p=2299420

**Comment by rkompass** [comment]
This is an old discussion topic. After some search I think you would need something like:

```
import os
from machine import UART

uart1 = machine.UART(1, 115200)

os.dupterm(None, 1)             # detach old stream (USB) from REPL
os.dupterm(uart1, 1)             # attach uart1 to REPL
```

see: [here](https://forum.micropython.org/viewtopic.php?t=10809) for example and [dupterm docs](https://docs.micropython.org/en/latest/library/os.html) and [here](https://docs.micropython.org/en/latest/esp8266/quickref.html#uart-serial-bus).
